src/react-apps/applications/shared/src/version-control/versionControlHeader.tsx

```diff
@@ -0,0 +1,392 @@
+import { createMuiTheme, createStyles, Grid, WithStyles, withStyles } from '@material-ui/core';
+import * as React from 'react';
+import altinnTheme from '../theme/altinnStudioTheme';
+import { getLanguageFromKey } from '../utils/language';
+import FetchChangesComponent from '../version-control/fetchChanges';
+import ShareChangesComponent from '../version-control/shareChanges';
+import SyncModalComponent from './syncModal';
+
+export interface IVersionControlHeaderProps extends WithStyles<typeof styles> {
+  language: any;
+}
+
+export interface IVersionControlHeaderState {
+  changesInMaster: boolean;
+  changesInLocalRepo: boolean;
+  moreThanAnHourSinceLastPush: boolean;
+  hasPushRight: boolean;
+  anchorEl: any;
+  modalState: any;
+  mergeConflict: boolean;
+}
+
+const theme = createMuiTheme(altinnTheme);
+
+const styles = createStyles({
+  headerStyling: {
+    background: theme.altinnPalette.primary.greyLight,
+    paddingTop: 10,
+  },
+});
+
+const initialState = {
+  header: '',
+  descriptionText: '',
+  isLoading: '',
+  shouldShowDoneIcon: false,
+  btnText: '',
+  shouldShowCommitBox: false,
+  btnMethod: '',
+};
+
+class VersionControlHeader extends React.Component<IVersionControlHeaderProps, IVersionControlHeaderState> {
+  public interval: any;
+  public _isMounted = false;
+  constructor(_props: IVersionControlHeaderProps) {
+    super(_props);
+    this.state = {
+      changesInMaster: false,
+      changesInLocalRepo: false,
+      moreThanAnHourSinceLastPush: false,
+      hasPushRight: false,
+      anchorEl: null,
+      mergeConflict: false,
+      modalState: initialState,
+    };
+  }
+
+  public performAPIFetch(url: string, callbackFunc: any, object?: any) {
+    fetch(url, object)
```

Comment:
I suggest you use Axios. Easier and less code: https://medium.com/@thejasonfile/fetch-vs-axios-js-for-making-http-requests-2b261cdd3af5

Axios is already in use. See shared / utils / networking.

Reply:
Updated to using axios